kv/send.go

```diff
@@ -243,19 +251,12 @@ var sendOneFn = sendOne
 //
 // Do not call directly, but instead use sendOneFn. Tests mock out this method
 // via sendOneFn in order to test various error cases.
-func sendOne(client batchClient, timeout time.Duration,
-	rpcContext *rpc.Context, done chan batchCall) {
+func sendOne(ctx context.Context, rpcContext *rpc.Context, client batchClient, done chan batchCall) {
 	addr := client.remoteAddr
 	if log.V(2) {
 		log.Infof("sending request to %s: %+v", addr, client.args)
 	}
 
-	// TODO(tamird/tschottdorf): pass this in from DistSender.
-	ctx := context.TODO()
-	if timeout != 0 {
-		ctx, _ = context.WithTimeout(ctx, timeout)
```

Comment:
i think this code should stay here, and that we should `defer cancel()` this context - it was an oversight not to have done that earlier